File: UI/mystagepass_admin/lib/models/CancelledEventItem/cancelled_event_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'cancelled_event_item.g.dart';

@JsonSerializable()
class CancelledEventItem {
  String? eventName;
  String? locationName;
  DateTime? eventDate;
  int? ticketsSold;
  int? refundsNeeded;
  double? totalRefundAmount;

  CancelledEventItem({
    this.eventName,
    this.locationName,
    this.eventDate,
    this.ticketsSold,
    this.refundsNeeded,
    this.totalRefundAmount,
  });

  factory CancelledEventItem.fromJson(Map<String, dynamic> json) =>
      _$CancelledEventItemFromJson(json);

  Map<String, dynamic> toJson() => _$CancelledEventItemToJson(this);
}

File: MyStagePass.Model/DTOs/CancelledEventItem.cs
using System;

namespace MyStagePass.Model.DTOs
{
	public class CancelledEventItem
	{
		public string EventName { get; set; }
		public string LocationName { get; set; }
		public DateTime EventDate { get; set; }
		public int TicketsSold { get; set; }
		public int RefundsNeeded { get; set; }
		public decimal TotalRefundAmount { get; set; }
	}
}
